pkgs/applications/editors/emacs/default.nix

```diff
@@ -67,11 +61,17 @@ stdenv.mkDerivation rec {
              "--with-gif=no" "--with-tiff=no" ])
     ++ lib.optional withXwidgets "--with-xwidgets";
 
-  NIX_CFLAGS_COMPILE = lib.optionalString (stdenv.isDarwin && withX)
-    "-I${cairo.dev}/include/cairo";
-
-  preBuild = ''
+  preConfigure = lib.optionalString srcRepo ''
+    ./autogen.sh
+  '' + lib.optionalString srcLocal ''
```

Comment:
Is it really necessary to control this via that argument to the expression? We could just run it all the time and simplify the expression by removing `srcLocal`